pkg/preference/implem.go

```diff
@@ -253,32 +254,23 @@ func (c *preferenceInfo) SetConfiguration(parameter string, value string) error
 		switch p {
 
 		case "timeout":
-			typedval, err := strconv.Atoi(value)
-			if err != nil {
-				return fmt.Errorf("unable to set %q to %q", parameter, value)
-			}
-			if typedval < 0 {
-				return errors.New("cannot set timeout to less than 0")
+			typedval, err := time.ParseDuration(value)
+			if err != nil || typedval < minimumDurationValue {
+				return fmt.Errorf("unable to set %q to %q, value must be a positive Duration (e.g. 4s, 5m, 1h); minimum value: 1s", parameter, value)
```

Comment:
Wondering if the error we return here should not also wrap the original `err` returned by `time.ParseDuration`, if any. This way, it would be reported to the user. What do you think?

Because if I enter a duration but with a unit suffix (like [days](https://github.com/golang/go/issues/17767)) not supported by `time.ParseDuration`, I find our error message not to be much helpful in understanding the issue:
```
❯ odo preference set PushTimeout 1d 
 ✗  unable to set "pushtimeout" to "1d", value must be a positive Duration (e.g. 4s, 5m, 1h); minimum value: 1s
```